src/controller/task.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response, NextFunction } from 'express'
+import TaskModel, { Task } from '../model/task'
+
+export default {
+  async createTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const { title, description, importance } = req.body
+      const newTask: Task = await TaskModel.create({
+        title,
+        description,
+        importance
+      })
+      res.status(201).json(newTask)
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getAllTasks (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const tasks: Task[] = await TaskModel.find()
+      res.status(200).json({
+        results: tasks.length,
+        tasks
+      })
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getTaskById (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const task: Task | null = await TaskModel.findById(req.params.id)
+      res.status(200).json(task)
+    } catch (err) {
+      next(err)
+    }
+  },
+  async updateTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const { title } = req.body
+      const updatedTask: Task = await TaskModel.findOneAndUpdate(
+        { _id: req.params.id },
+        { title: title },
+        { new: true })
+      res.status(200).json(updatedTask)
+    } catch (err) {
+      next(err)
+    }
+  },
+  async deleteTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      await TaskModel.findByIdAndDelete(req.params.id)
```

Comment:
`await TaskModel.deleteOne(...)` should work here, but again 404 should be if task is not found.

Reply:
I was thinking about moving null and not valid id checking to middleware (because this will be using in Get by id, Delete and Patch methods) ?